src/implementation/Server/HTTPServer/pubsub.ts

```diff
@@ -32,9 +32,14 @@ export default class HTTPServerPubSub implements IServerPubSub {
     await this.server.getServerImpl().registerPubSubSubscriptionRoute(pubsubName, topic, route);
 
     this.server.getServer().post(`/${route}`, async (req, res) => {
-      // Process our callback
       // @ts-ignore
-      await cb(req?.body?.data);
+      // Parse the data of the body, we prioritize fetching the data key in body if possible
+      // i.e. Redis returns { data: {} } and other services return {}
+      // @todo: This will be deprecated in an upcoming major version and only req.body will be returned
+      const data = req?.body?.data || req?.body;
+
+      // Process our callback
+      await cb(data);
```

Comment:
While we are at this, can cb throw an error? If yes, should we handle it and return an appropriate response?

Reply:
(this can be tracked separately too)

Reply:
https://github.com/dapr/js-sdk/issues/270

> (this can be tracked separately too)